testbed/tests/widgets/properties.py

```diff
@@ -320,9 +320,13 @@ async def test_font(widget, probe, verify_font_sizes):
     if verify_font_sizes[1]:
         assert probe.height > orig_height
 
-    # Reset to original family and size.
-    del widget.style.font_family
+    # Reset to original size.
     del widget.style.font_size
+    await probe.redraw(message="Widget text should be reset to original size")
+    probe.assert_font_size(SYSTEM_DEFAULT_FONT_SIZE)
+
+    # Reset to original family
+    del widget.style.font_family
```

Comment:
This clearly makes the test pass, but I think it is masking the underlying problem.

Running the test in `--slow` mode can be illustrative, as you can see the changes that are happening at a speed that is observable. 

In this case, using `--slow` highlights that there's a problem. If you revert your change to this test, you can see that after deleting the two font properties, the widget *is* still tall - that is, the test failure is reporting a problem correctly. 

If you change the *order* of the deletions - deleting the family, *then* the size - the widget height is correct, and the test passes.

So - introducing the extra probe redraw here actually isn't the fix. It's the fact that you've changed the order of the deletions.

Reply:
I made all kinds of changes to test_font() and although, it **visually** looked ok (with --slow), the test failed when checking the resetted widget height. Doing the same kind of changes in the selection example works, though. There, the widget height does not only look ok, Spinner.getHeight() in fact returns its original value.

So, if you've found a better way to make the testbed work, please apply these changes to this PR.

Reply:
I can't speak to your own personal setup - but the CI setup reveals the problem, and I can reproduce the problem locally, so this isn't an oddity of the testbed. There's a problem here that needs to be resolved.

What you're looking for is the amout of space *above* the text in the widget. The text has the right sized font, but there's a border *above* the widget (and presumably below as well), which effectively represents the old widget size, with the smaller text vertically centered in that space. 

As for a fix... you're the person proposing the change. It's up to you to fix problems that are found. I'm telling you by way of code review that the change you've made to the test suite isn't acceptable. Maybe @mhsmith has some more ideas about places to look for a fix, but we both have other development priorities (not the least of which is finally getting the next Toga release out the door), so we're not in a position to drop everything to hunt a bug that isn't a high priority for us personally.